### ISSUE TITLE
Reg\Transfers - Edit mode of Transfer form

### ISSUE THREAD

**Issue by andrea-williams** [bcgov/cas-registration#2447]
Blocked by #2427 , #1949 

#### Description:

As an internal user, I need to be able to edit a Transfer event if the status is "To be transferred" (i.e., the effective date of transfer is still in the future)

#### Acceptance Criteria:

Given I am an internal user 
When I look at a Transfer (#1713 ) with a status of "To be transferred"
Then it is in disabled mode, and I have the option to edit certain fields by clicking the "Edit Details" button
[View mode of To be Transferred Screen: Wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=25107-575&t=05j8hcBvabbp4bL1-4)

I also see a "Cancel transfer button"
Given that I clicked "Cancel transfer"
A confirmation modal pops up. 
[Wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=25249-8823&t=U3c01GvggMyuhRhK-4) 

Given that I clicked edit details
[Edit mode of To be Transferred Scren: Wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=27898-6188&t=vf2XdXjFsVZ2oWhA-4)
- [x] the form is enabled
- [x] the "edit details" button changes to "Transfer Entity" 

#### Development Checklist:

- [x] at the bottom of the page there is a "Cancel Transfer" button that will cancel the entire transfer event
- [x] fields that should be editable: "Effective date of transfer", Facilities dropdown (if applicable), Operation dropdown (if applicable)
- [x] fields that should NOT be editable: "Current operator", "Select the new operator", "What is being transferred?"
- [x] ~**to be decided by devs**: should "Select the new operation this facility will be allocated to" be editable or not?~ => Sep: I’m opting to skip this feature for now to keep the complexity low and save development time. However, it can be implemented later if it turns out to be an essential requirement.
- [ ] Meets the DOD

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by andrea-williams** [comment]
@zoeyli-46 it would probably be helpful if we were to have a separate wireframe for Transfers when they're being edited.

**Comment by andrea-williams** [comment]
thanks @zoeyli-46 ! Just noting that when the page is in Edit mode, we'll need to add another button so that users can entirely cancel the transfer if needed. But if you're super busy with other things don't worry about the wireframe - I think we can get by without it.

**Comment by zoeyli-46** [comment]
Added in AC x wireframe which includes cancelling the transfer button

**Comment by Sepehr-Sobhani** [comment]
Should we hide the edit button when the status is set to "Transferred"?

**Comment by andrea-williams** [comment]
> Should we hide the edit button when the status is set to "Transferred"?

definitely. Once the Transfer has taken place, editing should not be an option.

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://bcgov-my.sharepoint.com/:w:/g/personal/francesca_viduya_gov_bc_ca/Eb5eadR7dwJDjnrcPF6RiUsBbcgaHYwR98rprSg5SMa3TQ?e=COZ4yY).

I can verify that:
- [x] When viewing details for a transfer with a Status of 'To be transferred', the fields are disabled/not editable
- [x] When viewing details for a transfer with a Status of 'To be transferred', and the fields are in disabled mode, I see the buttons ```Edit Details```, ```Cancel Transfer```, and ```Back```
- [x] When transferring an operation, ```Edit Details``` button allows me to edit only the Operation and Effective date of transfer fields
- [x] When transferring a facility, ```Edit Details``` button allows me to edit only the Facilities and Effective date of transfer fields
- [x] When I click on ```Cancel Transfer```, [this confirmation modal](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=25249-8823) pops up 

A few things I would like to note:
- When editing the form, the ```Cancel Transfer``` button is still visible. This might be mistaken as a standard ```Cancel``` button, such as when one wants to get out of the current view. 
- I have an example on the test doc starting page 5 for the transfer of facilities. If I transfer two facilities on a single form, I will get two entries in the Transfers grid (page 7). Then, I click on ```View Details``` on one of the Facilities and click ```Cancel Transfer``` and continue with the cancellation, then both records will be deleted from the Transfers grid (Page 9) when I meant to cancel the transfer for just one of the facilities. Is this an expected behavior, or should there be just one record in the Transfers grid for both facilities?

cc: @patriciarussellCAS / @vesselak

**Comment by patriciarussellCAS** [comment]
Accepting this ticket.
I'll create a ticket for the design improvements suggested above.

**Comment by Sepehr-Sobhani** [comment]
Regarding:
> I have an example on the test doc starting page 5 for the transfer of facilities. If I transfer two facilities on a single form, I will get two entries in the Transfers grid (page 7). Then, I click on View Details on one of the Facilities and click Cancel Transfer and continue with the cancellation, then both records will be deleted from the Transfers grid (Page 9) when I meant to cancel the transfer for just one of the facilities. Is this an expected behavior, or should there be just one record in the Transfers grid for both facilities?

It was requested by the BA to allow transferring multiple facilities in one Transfer record, while displaying each facility in separate records in the table. This approach helps keep the grid clean and avoids clutter (Imagine showing the names of four facilities in a single row). Additionally, to cancel a single facility transfer, the user can edit the transfer and simply remove that facility. It’s technically not feasible (and misleading) to display only one facility in the form when the same transfer record includes multiple facilities.
